src/scheduler/mod.rs

```diff
@@ -27,12 +29,33 @@ pub struct Schedule {
     steps: Vec<ScheduleStep>,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
 enum ScheduleStep {
     Task(TaskId),
     Random,
 }
 
+/// A `ScheduleRecord` can be used to record both the step
+/// taken and all runnable tasks
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ScheduleRecord {
```

Comment:
I think we should move this into `determinism_check.rs`, since it's the only scheduler that's going to use it.